pkg/apis/config/validation/providerconfig_test.go

```diff
@@ -0,0 +1,216 @@
+				EgressFilter: &config.EgressFilter{
+					FilterListProviderType: config.FilterListProviderTypeStatic,
+				},
+			},
+			field.NewPath("config"),
+			ContainElement(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.filterListProviderType")})),
+			),
+		),
+		Entry("should return error for downloaderConfig in shoot config",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					DownloaderConfig: &config.DownloaderConfig{Endpoint: "https://example.com"},
+				},
+			},
+			field.NewPath("config"),
+			ContainElement(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.downloaderConfig")})),
+			),
+		),
+		Entry("should return error for ensureConnectivity in shoot config",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					EnsureConnectivity: &config.EnsureConnectivity{SeedNamespaces: []string{"foo"}},
+				},
+			},
+			field.NewPath("config"),
+			ContainElement(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.ensureConnectivity")})),
+			),
+		),
+		Entry("should return error if staticFilterList exceeds max entries",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					StaticFilterList: func() []config.Filter {
+						list := make([]config.Filter, 10001)
+						for i := range list {
+							list[i] = config.Filter{Network: "10.0.0.0/24", Policy: config.PolicyBlockAccess}
+						}
+						return list
+					}(),
+				},
+			},
+			field.NewPath("config"),
+			ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.staticFilterList")})),
+			),
+		),
+		Entry("should return error for invalid CIDR in staticFilterList",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					StaticFilterList: []config.Filter{
+						{Network: "invalid-cidr", Policy: config.PolicyBlockAccess},
+					},
+				},
+			},
+			field.NewPath("config"),
+			ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.staticFilterList.network")})),
+			),
+		),
+		Entry("should return error for invalid policy in staticFilterList",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					StaticFilterList: []config.Filter{
+						{Network: "10.0.0.0/24", Policy: "invalid-policy"},
+					},
+				},
+			},
+			field.NewPath("config"),
+			ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.egressFilter.staticFilterList.policy")})),
+			),
+		),
+		Entry("should succeed with valid StaticFilterList",
+			&config.Configuration{
+				EgressFilter: &config.EgressFilter{
+					StaticFilterList: []config.Filter{
+						{Network: "10.0.0.0/24", Policy: config.PolicyBlockAccess},
+						{Network: "2001:db8::/32", Policy: config.PolicyAllowAccess},
+					},
+				},
+			},
+			field.NewPath("config"),
+			BeEmpty(),
+		),
+		Entry("should return error for StaticFilterList with invalid CIDR",
```

Comment:
There is another test above (line 89) which seem to test the same thing. Do we need both?